### PR TITLE
fix: move from multiauth to rainbowkit

### LINKED ISSUE #247: Update the connected wallet when there's a wallet change event
**Description**
To prevent accidental mismatches between the wallet connected to the Cadastre and the address selected in the user's wallet, add listening for wallet change events and update the connected wallet accordingly.

### PR DESCRIPTION
**Pull request by tnrdd**
# Description

Move from `multiAuth` that use a library not maintained anymore to `rainbowKit`.
All the events are handled better and the library is being actively developed.

# Issue

fixes #247 

# Checklist:

- [x] My commit message follows the Conventional Commits specification
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have tested my code
- [x] My changes generate no new warnings
- [x] My PR is rebased off the most recent `develop` branch
- [x] My PR is opened against the `develop` branch

# Additional comments

- Upgrade to React v18 was needed to have `raibowKit` works without errors and warnings
- Remove unused libraries
- Remove unused files

# Alert Reviewers

@codynhat @gravenp

### CODE REVIEW

File: pages/_app.tsx [hunk 1]
```diff
@@ -7,30 +7,48 @@ import {
...
+import {
+  connectorsForWallets,
+  RainbowKitProvider,
+  lightTheme,
+} from "@rainbow-me/rainbowkit";
+import {
+  injectedWallet,
+  metaMaskWallet,
+  walletConnectWallet,
+  ledgerWallet,
```
Comment (gravenp) [inline pages/_app.tsx:25]: Can we add Coinbase Wallet and Brave Wallet as the second to last and last wallet options?

**Comment by gravenp** [comment]
@tnrdd this is looking really good. Handles the wallet address changes and signature rejections as we'd want.

Added one comment to add Coinbase & Brave as wallet options, but then this should be good to go!

**Review by gravenp — APPROVED** [review]
Looks good!

@codynhat Ready for your reivew.

File: components/ApproveOrPerformButton.tsx [hunk 2]
```diff
@@ -159,7 +159,6 @@ export function ApproveOrPerformButton(props: ApproveOrPerformButtonProps) {
         sender: account,
         providerOrSigner: sfFramework.settings.provider,
       });
-      console.log("HELLO1");
```
Comment (codynhat) [inline components/ApproveOrPerformButton.tsx:162]: Whoops. That's my bad.

**Review by codynhat — APPROVED** [review]
LGTM! Works great with Metamask and an injected wallet. Switching wallets works great and the saved DIDSession is working.

**Pull request merged**